test/demos/forwarddiffzero.jl

```diff
@@ -0,0 +1,76 @@
+"The simplest viable forward mode a AD, only supports `Float64`"
+module ForwardDiffZero
+
+using ChainRulesCore
+using Test
+
+struct Dual <: Real
+    primal::Float64
+    diff::Float64
+end
+
+primal(d::Dual) = d.primal
+diff(d::Dual) = d.diff
+
+primal(d::Real) = d
+diff(d::Real) = 0.0
+
+# needed for ^ to work from having `*` defined
+Base.to_power_type(x::Dual) = x
+
+
+function define_dual_overload(sig, ast)
+    opT, argTs = Iterators.peel(sig.parameters)
+    fieldcount(opT) == 0 || return  # not handling functors 
+    all(Float64 <: argT for argT in argTs) || return  # only handling purely Float64 ops.
+    op = opT.instance
+    opname = :($(parentmodule(op)).$(nameof(op)))
+    N = length(sig.parameters) - 1  # skip the op
+    fdef = quote
+        function $opname(dual_args::Vararg{Union{Dual, Float64},$N}; kwargs...)
+            ȧrgs = (NO_FIELDS,  diff.(dual_args)...)
+            args = ($opname, primal.(dual_args)...)
+            res = frule(ȧrgs, args...; kwargs...)
+            res === nothing && error("Apparently no rule for $($sig)), but we really thought there was, args=($args)")
+            y, ẏ = res
+            return Dual(y, ẏ)  # if y, ẏ are not `Float64` this will error.
+        end
+    end
+    # @show fdef
+    @eval $fdef
+end
+
+#########################################
+# Initial rule setup
+@scalar_rule x + y (One(), One())
```

Comment:
```suggestion
@scalar_rule x + y (1, 1)
```
because `One()` is bad.